lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
+
+    def initialize(*args)
+      super
+      tr!('/', '-')
+    end
+
+    # @return [String] The slash separated module name.
+    def forge_name
+      String.new(self).tr('-', '/')
+    end
```

Comment:
Doing a `#tr` directly will create a new ModuleName object that has a forward slash separator, which isn't want I would expect since the class tries to reverse that on creation - so I think returning an explicit string is a bit more complexity but less surprising in behavior.

```
[1] pry(#<RSpec::ExampleGroups::PuppetForgeModuleName>)> subject.to_path
=> "authorname/modulename"
[2] pry(#<RSpec::ExampleGroups::PuppetForgeModuleName>)> subject.to_path.class
=> PuppetForge::ModuleName
```

What do you think? Go with `#tr` or do the explicit clone/modify?